### PR TITLE
[mxnet18][neuron] - Add neuron mxnet18 based dlc image

### PR DESCRIPTION
**Pull request by aws-vrnatham**
*Issue #, if available:*

## PR Checklist
- [ x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Checklist
* When creating a PR:
- [ x] I've modified `src/config/build_config.py` in my PR branch by setting `ENABLE_EI_MODE = True` or `ENABLE_NEURON_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above
#### Benchmark Checklist
* When creating a PR:
- [ ] I've modified `src/config/test_config.py` in my PR branch by setting `ENABLE_BENCHMARK_DEV_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [ ] I've verified the code change on the config file mentioned above has already been reverted

*Description:*

*Tests run:*

*DLC image/dockerfile:*

*Additional context:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: mxnet/inference/docker/1.8.0/py3/Dockerfile.neuron [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+ENV LANG C.UTF-8
+ENV LD_LIBRARY_PATH /opt/conda/lib/:$LD_LIBRARY_PATH
+ENV PATH /opt/conda/bin:$PATH
+ENV TEMP=/home/model-server/tmp
+
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends software-properties-common \
+    && add-apt-repository ppa:openjdk-r/ppa \
+    && apt-get update \
+    && apt-get install -y --no-install-recommends \
```
Comment (saimidu) [inline mxnet/inference/docker/1.8.0/py3/Dockerfile.neuron:31]:
```suggestion
 && apt-get install -y --no-install-recommends software-properties-common \
 && add-apt-repository ppa:openjdk-r/ppa \
 && apt-get update \
 && apt-get install -y --no-install-recommends \
```

File: mxnet/inference/docker/1.8.0/py3/Dockerfile.neuron [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    libcap-dev \
+    gpg-agent
+
+RUN echo "deb https://apt.repos.neuron.amazonaws.com xenial main" > /etc/apt/sources.list.d/neuron.list
+RUN wget -qO - https://apt.repos.neuron.amazonaws.com/GPG-PUB-KEY-AMAZON-AWS-NEURON.PUB | apt-key add -
+RUN apt-get update
+
+RUN apt-get install -y \
+    aws-neuron-runtime \
+    aws-neuron-tools
```
Comment (saimidu) [inline mxnet/inference/docker/1.8.0/py3/Dockerfile.neuron:59]:
```suggestion
RUN apt-get update \
 && apt-get install -y \
    aws-neuron-runtime \
    aws-neuron-tools \
 && rm -rf /var/lib/apt/lists/* \     
 && apt-get clean
```

File: mxnet/inference/docker/1.8.0/py3/Dockerfile.neuron [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    libsm6 \
+    libxext6 \
+    libxrender-dev \
+    openjdk-11-jdk \
+    vim \
+    wget \
+    unzip \
+    zlib1g-dev \
+    libcap-dev \
+    gpg-agent
```
Comment (saimidu) [inline mxnet/inference/docker/1.8.0/py3/Dockerfile.neuron:51]:
```suggestion
    gpg-agent \
 && rm -rf /var/lib/apt/lists/* \     
 && apt-get clean
```

**Comment by saimidu** [comment]
For some reason, the MX 1.8 images don't seem to have been built on the PR. Taking a look at this.

**Comment by aws-vrnatham** [comment]
> For some reason, the MX 1.8 images don't seem to have been built on the PR. Taking a look at this.

Please let me know what needs to be done to get the test working.

File: mxnet/inference/docker/1.8.0/py3/Dockerfile.neuron [hunk 4]
```diff
@@ -0,0 +1,164 @@
+FROM ubuntu:18.04
```
[the remark this answers is not in the document]
Comment (aws-vrnatham) [reply mxnet/inference/docker/1.8.0/py3/Dockerfile.neuron]: Done

File: mxnet/buildspec-neuron.yml [hunk 5]
```diff
@@ -1,8 +1,8 @@
   account_id: &ACCOUNT_ID <set-$ACCOUNT_ID-in-environment>
   region: &REGION <set-$REGION-in-environment>
   framework: &FRAMEWORK mxnet
-  version: &VERSION 1.5.1
-  os_version: &OS_VERSION ubuntu16.04
+  version: &VERSION 1.8.0
+  os_version: &OS_VERSION ubuntu18.04
```
[the remark this answers is not in the document]
Comment (aws-vrnatham) [reply mxnet/buildspec-neuron.yml:5]: Done

File: mxnet/inference/docker/1.8.0/py3/Dockerfile.neuron [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+LABEL dlc_major_version="1"
+
+# Specify accept-bind-to-port LABEL for inference pipelines to use SAGEMAKER_BIND_TO_PORT
+# https://docs.aws.amazon.com/sagemaker/latest/dg/inference-pipeline-real-time.html
+LABEL com.amazonaws.sagemaker.capabilities.accept-bind-to-port=true
+# Specify multi-models LABEL to indicate container is capable of loading and serving multiple models concurrently
+# https://docs.aws.amazon.com/sagemaker/latest/dg/build-multi-model-build-container.html
+LABEL com.amazonaws.sagemaker.capabilities.multi-models=true
+
+ARG MMS_VERSION=1.1.2
```
[the remark this answers is not in the document]
Comment (aws-vrnatham) [reply mxnet/inference/docker/1.8.0/py3/Dockerfile.neuron]: Done

File: mxnet/inference/docker/1.8.0/py3/Dockerfile.neuron [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+    zlib1g-dev \
+    libcap-dev \
+    gpg-agent \
+  && rm -rf /var/lib/apt/lists/* \
+  && rm -rf /tmp/tmp* \
+  && apt-get clean
+
+RUN echo "deb https://apt.repos.neuron.amazonaws.com xenial main" > /etc/apt/sources.list.d/neuron.list
+RUN wget -qO - https://apt.repos.neuron.amazonaws.com/GPG-PUB-KEY-AMAZON-AWS-NEURON.PUB | apt-key add -
+RUN apt-get update
```
[the remark this answers is not in the document]
Comment (aws-vrnatham) [reply mxnet/inference/docker/1.8.0/py3/Dockerfile.neuron]: Done

File: mxnet/inference/docker/1.8.0/py3/Dockerfile.neuron [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+
+LABEL maintainer="Amazon AI"
+LABEL dlc_major_version="1"
+
+# Specify accept-bind-to-port LABEL for inference pipelines to use SAGEMAKER_BIND_TO_PORT
+# https://docs.aws.amazon.com/sagemaker/latest/dg/inference-pipeline-real-time.html
+LABEL com.amazonaws.sagemaker.capabilities.accept-bind-to-port=true
+# Specify multi-models LABEL to indicate container is capable of loading and serving multiple models concurrently
+# https://docs.aws.amazon.com/sagemaker/latest/dg/build-multi-model-build-container.html
+LABEL com.amazonaws.sagemaker.capabilities.multi-models=true
```
[the remark this answers is not in the document]
Comment (aws-vrnatham) [reply mxnet/inference/docker/1.8.0/py3/Dockerfile.neuron]: Yes. Even though mme does not take into account yet mem fails in neuron etc atleast it should work for cpu cases

File: mxnet/buildspec-neuron.yml [hunk 9]
```diff
@@ -17,14 +17,14 @@
...
-        target:  mms-entrypoint.py
+        source: docker/artifacts/entrypoint.sh
+        target:  entrypoint.sh
 
   images:
     BuildNeuronMXNetInferencePy3DockerImage:
       <<: *INFERENCE_REPOSITORY
       build: &MXNET_INF_INFERENCE_PY3 false
-      image_size_baseline: 5000
+      image_size_baseline: 10000
```
[the remark this answers is not in the document]
Comment (aws-vrnatham) [reply mxnet/buildspec-neuron.yml:28]: After the change it is coming out to 5111MB. Will having the size of 10000 cause any issues.

File: mxnet/inference/docker/1.8.0/py3/Dockerfile.neuron [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+  && rm -rf /var/lib/apt/lists/* \
+  && rm -rf /tmp/tmp* \
+  && apt-get clean
+
+# https://github.com/docker-library/openjdk/issues/261 https://github.com/docker-library/openjdk/pull/263/files
+RUN keytool -importkeystore -srckeystore /etc/ssl/certs/java/cacerts -destkeystore /etc/ssl/certs/java/cacerts.jks -deststoretype JKS -srcstorepass changeit -deststorepass changeit -noprompt; \
+    mv /etc/ssl/certs/java/cacerts.jks /etc/ssl/certs/java/cacerts; \
+    /var/lib/dpkg/info/ca-certificates-java.postinst configure;
+
+RUN curl -L -o ~/miniconda.sh https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh \
```
[the remark this answers is not in the document]
Comment (aws-vrnatham) [reply mxnet/inference/docker/1.8.0/py3/Dockerfile.neuron]: Removed it.

File: mxnet/buildspec-neuron.yml (hunk 9 above)
[the remark this answers is not in the document]
Comment (saimidu) [reply mxnet/buildspec-neuron.yml:28]: It wouldn't allow us to catch unexpected size increases.

**Review by saimidu — COMMENTED** [review]
Approved. Reverting config changes.

**Review by saimidu — APPROVED** [review]

**Pull request merged**